src/main/java/com/thirdcc/webapp/web/rest/FinanceReportResource.java

```diff
@@ -0,0 +1,79 @@
+package com.thirdcc.webapp.web.rest;
+
+import com.thirdcc.webapp.domain.enumeration.TransactionType;
+import com.thirdcc.webapp.service.FinanceReportService;
+import com.thirdcc.webapp.service.YearSessionService;
+import com.thirdcc.webapp.service.dto.FinanceReportDTO;
+import com.thirdcc.webapp.utils.YearSessionUtils;
+import io.github.jhipster.web.util.PaginationUtil;
+import io.github.jhipster.web.util.ResponseUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.math.BigDecimal;
+import java.time.Month;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api")
+public class FinanceReportResource {
+    private final Logger log = LoggerFactory.getLogger(FinanceReportResource.class);
+
+    private static final String ENTITY_NAME = "financeReport";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final FinanceReportService financeReportService;
+
+    private final YearSessionService yearSessionService;
+
+    public FinanceReportResource(FinanceReportService financeReportService, YearSessionService yearSessionService) {
+        this.financeReportService = financeReportService;
+        this.yearSessionService = yearSessionService;
+    }
+
+    @GetMapping("/finance-report")
+    public ResponseEntity<List<FinanceReportDTO>> getAllEventFinanceReport(
+        Pageable pageable,
+        @RequestParam MultiValueMap<String, String> queryParams,
+        UriComponentsBuilder uriBuilder
+    ) {
+        log.debug("REST request to get a page of FinanceReport");
+        Page<FinanceReportDTO> page = financeReportService.findAll(pageable);
+        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(uriBuilder.queryParams(queryParams), page);
+        return ResponseEntity.ok().headers(headers).body(page.getContent());
+    }
+
+    @GetMapping("/finance-report/event/{eventId}")
+    public ResponseEntity<FinanceReportDTO> getFinanceReportByEventId(@PathVariable Long eventId) {
+        log.debug("REST request to get FinanceReport by eventId: {}", eventId);
+        Optional<FinanceReportDTO> financeReportDTO = financeReportService.findOneByEventId(eventId);
+        return ResponseUtil.wrapOrNotFound(financeReportDTO);
+    }
+
+    @GetMapping("/finance-report/year-session")
```

Comment:
yes. This is currently how it works.

you have better suggestion? I prefer not to overload the `api/finance-report` endpoint tho

Reply:
My suggestion is change the url to:
api/finance-report/yearly?year-session=2019/2020
Or
api/finance-report/year-session?value=2019/2020

Reply:
sorry. my mistake. The API looks like this
`/finance-report/year-session?yearSessionId=1`
 where yearSessionId is referring to the primary key of `year_session` table.

The reason is I'm thinking the UI will have a dropdown button (where we load all dropdown options from `GET` all year_session table) then the user can switch between year_session by providing the id. I believe the f/e should not give "2019/2020" as query param because the probability of having typo is high.

if no query param is given, the last record in the  `year_session` table will be chosen. 
If no record is found in `year_session` table, a 500 error will be thrown.

Reply:
i agree user will not fill in the year session in FE
but using yearSessionId in params is good?
will it diff for use when we want to debug?
actually I dun really feel the Id column in yearSession is important. 
Initial i want to have something like the jhi_authority table

Reply:
we can discuss this on Friday night meeting

Reply:
ok

Reply:
Use Id as Request Param